exchanges/kucoin/kucoin_wrapper.go

```diff
@@ -0,0 +1,1349 @@
+			TransferType: "deposit",
+			CryptoTxID:   depositsData.Items[x].WalletTxID,
+			Status:       depositsData.Items[x].Status,
+			Amount:       depositsData.Items[x].Amount,
+			Currency:     depositsData.Items[x].Currency,
+		}
+	}
+	length := len(depositsData.Items)
+	for x := range withdrawalsData.Items {
+		fundingData[length+x] = exchange.FundHistory{
+			Fee:             withdrawalsData.Items[x].Fee,
+			Timestamp:       withdrawalsData.Items[x].UpdatedAt.Time(),
+			ExchangeName:    ku.Name,
+			TransferType:    "withdrawal",
+			CryptoToAddress: withdrawalsData.Items[x].Address,
+			CryptoTxID:      withdrawalsData.Items[x].WalletTxID,
+			Status:          withdrawalsData.Items[x].Status,
+			Amount:          withdrawalsData.Items[x].Amount,
+			Currency:        withdrawalsData.Items[x].Currency,
+			TransferID:      withdrawalsData.Items[x].ID,
+		}
+	}
+	return fundingData, nil
+}
+
+// GetWithdrawalsHistory returns previous withdrawals data
+func (ku *Kucoin) GetWithdrawalsHistory(ctx context.Context, c currency.Code, a asset.Item) ([]exchange.WithdrawalHistory, error) {
+	var err error
+	switch a {
+	case asset.Spot:
+		var withdrawals *HistoricalDepositWithdrawalResponse
+		withdrawals, err = ku.GetHistoricalWithdrawalList(ctx, c.String(), "", time.Time{}, time.Time{}, 0, 0)
+		if err != nil {
+			return nil, err
+		}
+		resp := make([]exchange.WithdrawalHistory, len(withdrawals.Items))
+		for x := range withdrawals.Items {
+			resp[x] = exchange.WithdrawalHistory{
+				Status:       withdrawals.Items[x].Status,
+				CryptoTxID:   withdrawals.Items[x].WalletTxID,
+				Timestamp:    withdrawals.Items[x].CreatedAt.Time(),
+				Amount:       withdrawals.Items[x].Amount,
+				TransferType: "withdrawal",
+				Currency:     c.String(),
+			}
+		}
+		return resp, nil
+	case asset.Futures:
+		var futuresWithdrawals *FuturesWithdrawalsListResponse
+		futuresWithdrawals, err = ku.GetFuturesWithdrawalList(ctx, c.String(), "", time.Time{}, time.Time{})
+		if err != nil {
+			return nil, err
+		}
+		resp := make([]exchange.WithdrawalHistory, len(futuresWithdrawals.Items))
+		for y := range futuresWithdrawals.Items {
+			resp[y] = exchange.WithdrawalHistory{
+				Status:       futuresWithdrawals.Items[y].Status,
+				CryptoTxID:   futuresWithdrawals.Items[y].WalletTxID,
+				Timestamp:    futuresWithdrawals.Items[y].CreatedAt.Time(),
+				Amount:       futuresWithdrawals.Items[y].Amount,
+				Currency:     c.String(),
+				TransferType: "withdrawal",
+			}
+		}
+		return resp, nil
+	default:
+		return nil, fmt.Errorf("%w %v", asset.ErrNotSupported, a)
+	}
+}
+
+// GetRecentTrades returns the most recent trades for a currency and asset
+func (ku *Kucoin) GetRecentTrades(ctx context.Context, p currency.Pair, assetType asset.Item) ([]trade.Data, error) {
+	var resp []trade.Data
+	const limit = 1000
+	switch assetType {
+	case asset.Futures:
+		p.Delimiter = ""
+		tradeData, err := ku.GetFuturesTradeHistory(ctx, p.String())
```

Comment:
Futures orders don't have delimiters or the delimiter is ""(empty string). and Since setting an empty string as a delimiter is not allowed either in the wrapper or config file is not allowed I used an underscore delimiter as a delimiter and re-assigned "" empty string when converting to String representation from currency Pair.